enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,709 @@
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: MachineAutoscaler ready for autoscaling
+      reason: MachineAutoscalerReady
+      status: "True"
+      type: Ready
+```
+
+In a case where the user has specified the same MachineSet in multiple MachineAutoscaler resources,
+the condition would appear as follows:
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: targetScaleRef has multiple MachineAutoscaler owners
+      reason: MachineAutoscalerDuplicateTargetScaleRef
+      status: "False"
+      type: Ready
+```
+
+In this manner, errors in user configuration will be exposed through the MachineAutoscaler
+resource to assist users in diagnosing error states.
+
+#### MachineSet Sync Controller Changes
+
+The behavior of the MachineSet sync controller is another focus of implementation detail.
+Specifically, the MachineSet annotations and the replicas field of the spec
+(i.e. `.spec.replicas`). As mentioned previously, the MachineSet sync controller will
+use the managed fields metadata to know who, or what, has updated a field. Based on the field,
+and the author of the update, the sync controller will either propagate the change
+or synchronize with the authoritative resource.
+
+The updates that the MachineSet sync controller will watch for fall into a few different
+categories:
+
+* `.metadata.annotations` changes will be synced from the authoritative resource to the
+  non-authoritative except in these cases:
+  * The Cluster Autoscaler Operator has added the minimum and maximum size annotations, and ownership
+    annotation to a record. If the sync controller sees an update to these annotations on a
+    non-authoritative resource originating from the Cluster Autoscaler Operator, it will copy
+    that change to the authoritative resource if no MachineAutoscaler is referencing the
+    authoritative resource.
+  * A provider MachineSet controller has added the scale from zero annotations to a
+    non-authoritative record. This occurs when the Cluster API resource is marked as
+    authoritative but the Machine API resource is updated by the provider MachineSet controller.
+    In these cases the scale from zero annotations will be copied to the non-authoritative
+    Cluster API resource. The data from the MachineSet controller is only applied to
+    Machine API resources currently.
+* `.spec.replicas` changes will be synced from the Cluster API MachineSet to the Machine
+  API MachineSet regardless of which is authoritative when the change originates from the
+  Cluster Autoscaler. As the Cluster Autoscaler will be configured to operate against Cluster
+  API resources only, there will be a need to identify when the Cluster Autoscaler has updated
+  a non-authoritative Cluster API resource so that the authoritative resource can be updated.
+  This will only occur when the sync controller observes and update to the replicas field from
+  the Cluster Autoscaler.
```

Comment:
We intend to use a VAP to block updates to a non-authoritative resource, have you investigated how we let the CAS through the VAP (what exception do we need in place) to make just this replicas field change?

Reply:
i have not, do you happen to have a link where i could learn more?

Reply:
Our intention is to use VAP (https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/) to block writes to the object unless they come specifically from sync controller, or, the resource itself is authoritative.

I think what would be interesting to explore would be if we could add an exception allowing another service account to write, but specifically only to the status subresource

Reply:
yeah, that is interesting.

Reply:
i'm still researching the vap details here.

Reply:
thanks Theo! i've added some example VAPs but i could use some help to make them better.